### PR TITLE
Return all trial's sources metadata in trials.sources object

### PR DESCRIPTION
**Pull request by vitorbaptista**
See https://github.com/opentrials/opentrials/issues/327 for the reasoning behind this.

### CODE REVIEW

**Comment by nightsh** [comment]
Some comments:

* Shouldn't we remove the `source_id` from the Trial model?
* There are some schema validation errors (trying now on `/search` endpoint)

File: api/models/publication.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ const Publication = BaseModel.extend({
       id: attributes.id,
       url: attributes.url,
       title: attributes.title,
-      source: this.related('source').toJSON(),
+      source_id: this.attributes.source_id,
```
Comment (nightsh) [inline api/models/publication.js]: Why not `attributes.source_id`? Same for `record.js`

Comment (vitorbaptista) [inline api/models/publication.js]: This is a bit messy :/

`attributes` here is `this.toJSON()`. However, it'll only contain the model's visible attributes. `source_id` isn't one of them, so we need to get directly from `this.attributes`

**Comment by vitorbaptista** [comment]
> * Shouldn't we remove the `source_id` from the Trial model?

Not now, as it's important to know which actual source we're displaying info from. We'll end up removing it only when (and if) we start merging attributes from multiple sources.

> * There are some schema validation errors (trying now on /search endpoint)

As we talked, you need to run `npm run reindex`

File: api/models/publication.js (hunk 1 above)
Comment (nightsh) [inline api/models/publication.js]: I was expecting something like that, but I thought I'd double check with you before testing.

Why can't we just do `const attributes = this.attributes`? Might be a silly question, but I didn't get to run it yet.

**Comment by nightsh** [comment]
All good, you can merge it as far as I'm concerned 👍

**Pull request merged**